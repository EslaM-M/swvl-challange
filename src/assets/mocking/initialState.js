// import route from "../data/route";
import tripInformation from "../data/trip-information";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
