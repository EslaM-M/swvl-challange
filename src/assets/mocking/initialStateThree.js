// import route from "../data/route";
import tripInformation from "../data/trip-information";
import fullBooking from "../data/full-booking-list";
export const initialStateThree = {
  bookings: fullBooking,
  bookingAllowed: true,
  route: tripInformation.route.map(e => {
    e.count = 0;
    return e;
  }),
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
