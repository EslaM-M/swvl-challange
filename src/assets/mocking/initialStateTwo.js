// import route from "../data/route";
import tripInformation from "../data/trip-information";
import bookings from "../data/booking";
export const initialState = {
  bookings: [],
  bookingAllowed: true,
  route: tripInformation.route,
  bookings: bookings,
  captainLocation: {
    lat: null,
    lng: null
  },
  tripStatus: "NOT_STARTED",
  tripStartTime: null,
  arrivalStatus: "",
  currentStationId: null,
  fullDistance: null,
  baseFare: tripInformation.baseFare,
  startPoint: tripInformation.startPoint,
  endPoint: tripInformation.endPoint,
  driver: tripInformation.driver
};
