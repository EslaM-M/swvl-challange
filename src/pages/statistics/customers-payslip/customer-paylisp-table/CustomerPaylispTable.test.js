import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CustomerPaylispTable from "./index";
import { initialState } from "../../../../assets/mocking/initialStateTwo";
describe("CustomerPaylispTable Component", () => {
  let realUseContext;
  let useContextMock;
  // Setup mock
  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
    useContextMock.mockReturnValue({
      state: initialState
    });
  });
  // Cleanup mock
  afterEach(() => {
    React.useContext = realUseContext;
  });

  test("Component snapshot render correctly", () => {
    const container = render(<CustomerPaylispTable />);
    expect(container).toMatchSnapshot();
  });

  test("Component show the same number of booking in the state", async () => {
    const { baseElement } = render(<CustomerPaylispTable />);
    const bookings = baseElement.querySelectorAll("tr");

    expect(bookings.length).toBe(3);
  });
  test("Component show zero number of booking if there is no booking", async () => {
    initialState.bookings = [];
    useContextMock.mockReturnValue({
      state: initialState
    });
    const { baseElement } = render(<CustomerPaylispTable />);

    const bookings = baseElement.querySelectorAll("tr");
    expect(bookings.length).toBe(1);
  });
});
