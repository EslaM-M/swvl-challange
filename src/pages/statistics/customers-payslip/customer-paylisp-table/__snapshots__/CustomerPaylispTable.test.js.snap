// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerPaylispTable Component Component snapshot render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="table-responsive"
        >
          <table
            class="table"
          >
            <thead>
              <tr>
                <th>
                  #
                </th>
                <th>
                  Code
                </th>
                <th>
                  Name
                </th>
                <th>
                  Payment Method
                </th>
                <th>
                  Paid
                </th>
                <th>
                  Pickup Station
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  data-testid="index"
                >
                  1
                </td>
                <td
                  data-testid="user-id"
                >
                  936
                </td>
                <td
                  data-testid="user-name"
                >
                  Hossam Hassan
                </td>
                <td
                  data-testid="payment-type"
                >
                  credit
                </td>
                <td>
                  <label
                    data-testid="price"
                  >
                    45
                  </label>
                  <label
                    data-testid="currency"
                  >
                    EGP
                  </label>
                </td>
                <td
                  data-testid="station"
                >
                  Station
                  5
                </td>
              </tr>
              <tr>
                <td
                  data-testid="index"
                >
                  2
                </td>
                <td
                  data-testid="user-id"
                >
                  937
                </td>
                <td
                  data-testid="user-name"
                >
                  Hossam Hassan
                </td>
                <td
                  data-testid="payment-type"
                >
                  credit
                </td>
                <td>
                  <label
                    data-testid="price"
                  >
                    45
                  </label>
                  <label
                    data-testid="currency"
                  >
                    EGP
                  </label>
                </td>
                <td
                  data-testid="station"
                >
                  Station
                  5
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="table-responsive"
      >
        <table
          class="table"
        >
          <thead>
            <tr>
              <th>
                #
              </th>
              <th>
                Code
              </th>
              <th>
                Name
              </th>
              <th>
                Payment Method
              </th>
              <th>
                Paid
              </th>
              <th>
                Pickup Station
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                data-testid="index"
              >
                1
              </td>
              <td
                data-testid="user-id"
              >
                936
              </td>
              <td
                data-testid="user-name"
              >
                Hossam Hassan
              </td>
              <td
                data-testid="payment-type"
              >
                credit
              </td>
              <td>
                <label
                  data-testid="price"
                >
                  45
                </label>
                <label
                  data-testid="currency"
                >
                  EGP
                </label>
              </td>
              <td
                data-testid="station"
              >
                Station
                5
              </td>
            </tr>
            <tr>
              <td
                data-testid="index"
              >
                2
              </td>
              <td
                data-testid="user-id"
              >
                937
              </td>
              <td
                data-testid="user-name"
              >
                Hossam Hassan
              </td>
              <td
                data-testid="payment-type"
              >
                credit
              </td>
              <td>
                <label
                  data-testid="price"
                >
                  45
                </label>
                <label
                  data-testid="currency"
                >
                  EGP
                </label>
              </td>
              <td
                data-testid="station"
              >
                Station
                5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
